py_ecc/bls/point_compression.py

```diff
@@ -161,4 +230,12 @@ def decompress_G2(p: G2Compressed) -> G2Uncompressed:
         raise ValueError(
             "The given point is not on the twisted curve over FQ**2"
         )
+
+    # Validate z2 flags
+    c_flag2 = get_c_flag(z2)
+    b_flag2 = get_b_flag(z2)
+    a_flag2 = get_a_flag(z2)
```

Comment:
You might wrap this up into something like `c_flag2, b_flag2, a_flag2 = get_flags(z2)`